Dco dataset and distribution

mrpatrickwest opened the pull request:
custom displays, custom forms, filegraph for datasets and distributions

Commit:
Visualizing accessURL, downloadURL and landingPage

Commit:
Adding foaf:Document

for landingPage accessURL, downloadURL

Commit:
Removed debug stmts going to catalina.out

Left over from when creating the configuration parameters for DCO specific code

Commit:
Editing accessURL, downloadURL, landingPage

Ability to create, edit and delete the accessURL and downloadURL of a distribution and the landingPage of a dataset

Commit:
More documentation on custom forms

To help us in the future for creating custom forms I've added more inline comments in the classes

Commit:
Additional documentation on creating URL props

accessURL, downloadURL and landingPage ftl updated adding more comments to help future development

zednis commented:
This is on dco.tw or udco.tw?

zednis commented on src/edu/cornell/mannlib/vitro/webapp/edit/n3editing/configuration/generators/DatasetLandingPageGenerator.java:
```diff
@@ -0,0 +1,205 @@
+
+    // if we had any new resources, like creating a new object or a new sub object, then we would do that here
+    private Map<String, String> generateNewResources(VitroRequest vreq) {
+        String DEFAULT_NS_TOKEN=null; //null forces the default NS
```
This String DEFULAT_NS_TOKEN isn't used anywhere, is it?

edit: we should be able to remove this line.  We could even update the code that calls generateNewResources( ) and have it call `new HashMap<>();` directly.

mrpatrickwest replied:
DEFULAT_NS_TOKEN is not used in this case. Can be removed.

The only reason I kept the methods there is because we'll need them in other custom forms. Though we could just look at other code for that, like our project update code. In fact, we can just not call that method in getEditConfiguration. I'll take a look.

zednis replied:
I actually want to dramatically refactor the implementation of `getEditConfiguration( )`.  The implementation as it is calls a bunch of void methods with basically useless method parameters.  It's hard to read because the method return types and parameters don't mean much.

...but I think we should do that in another ticket.

zednis commented:
I want to refactor the implementation of `getEditConfiguration( )` in our extensions to VivoBaseGenerator, but that should be a different ticket.

mrpatrickwest commented:
This is on udco. The code is for accessURL and downloadURL on a Distribution. And landingPage on Dataset. Create a dcodata:Dataset and then a dcodata:Distribution. When you create an accessURL, downloadURL or landingPage you should be able to just enter a URL and when it returns to the Dataset or Distribution you'll see that link. If you try to edit it it'll tell you to add a new one and delete the old one. I felt it was way too complicated to delete the foaf:Document with the old URI and then create a new foaf:Document with the new URI.

zednis commented:
Under overview on the dcodata:Dataset profile page I see 2 properties called "dataset page" which appear to mean the landing page for the dataset.

<img width="894" alt="screen shot 2016-05-16 at 4 49 42 pm" src="https://cloud.githubusercontent.com/assets/1057295/15306245/3ba04eba-1b86-11e6-99bc-89a0ac278344.png">

The (+) option for both links to https://udco.tw.rpi.edu/vivo/editRequestDispatch?subjectUri=http%3A%2F%2Finfo.deepcarbon.net%2Findividual%2Fn19031&predicateUri=http%3A%2F%2Fwww.w3.org%2Fns%2Fdcat%23landingPage&domainUri=http%3A%2F%2Fwww.w3.org%2Fns%2Fdcat%23Dataset&rangeUri=http%3A%2F%2Fxmlns.com%2Ffoaf%2F0.1%2FDocument

the `rdfs:label` for `dcat:landingPage` is "landing page"@en - so shouldn't this be "landing page" in the UI?

Commit:
Removed un-needed methods and fixed ftl comment

Since we aren't generating any new resource uri's just removed the methods in the Generator classes. There was also a comment in addDownloadURL.ftl that was
missing the ending -->. And the wording in addLandingPage.ftl was "To edit an dataset ...", changed that to "To edit a dataset".

zednis commented:
After creating the landing page and going back to the dataset profile page I only see 1 entry for "dataset page" and it has a link to the landing page URL I entered.  Clicking the link attempted to open a new tab with the landing page URL - so that looks good.

I created a 2nd test dataset and got 2 "dataset page" properties when the profile page first loaded.  @mrpatrickwest can you see this behavior when you create a dataset instance?

zednis commented:
I still see 2 "dataset page" properties on https://udco.tw.rpi.edu/vivo/display/n14146 (I was not sure if the most recent update was an attempt to fix that)

<img width="884" alt="screen shot 2016-05-16 at 7 07 06 pm" src="https://cloud.githubusercontent.com/assets/1057295/15308345/6ea88332-1b99-11e6-8c95-e0baf771127d.png">

Commit:
Changed label of dcodata:Dataset to just Dataset

Was DCO Dataset

Commit:
For now not uploading file so disabling servlet

Not sure what these do, but for now commented out

Commit:
This is not any different from the vitro version

After removing the commented out code and the code that was creating the ckan distribution the file is no different from the vitro version, so removing

Commit:
Hardcoded endpoint to production fuseki

Changed to get the endpoint from ServerInfo so can point locally.

mrpatrickwest commented:
Still doesn't fix the two dataset page properties. Weird part is that it goes away after you create a dataset page.

And the reason I changed it to dataset page is that no one understood landing page. No one meaning Secretariat, Engagement or Elizabeth Ferriss.

zednis commented:
But that means we have a "dataset page" link on the dataset profile page - which I expect people will shorten to just calling the dataset page.

mrpatrickwest commented:
Hmm, I see your point ... suggestions on what to call it? It's the... well ... external landing page for the dataset. If it's data that's being uploaded by us then it might be considered the dataset summary page in drupal or the instance page itself in VIVO

zednis commented:
"external dataset webpage" ?  Long, but explicit.

Commit:
Creation of distributions allowed on property

Also removed the dcterms label since we're using dcterms directly now.

Commit:
Changing label of landing page to external dataset

zednis commented on productMods/templates/freemarker/body/partials/individual/propStatement-dcat-landingPage.ftl:
```diff
@@ -0,0 +1,13 @@
+<@showStatement statement />
+
+<#macro showStatement statement>
+    <a href="${statement.value!}" title="Dataset Page" target="_blank">${statement.value!"dataset page not found"}</a>
```
Add 'external' to the title here?

mrpatrickwest commented:
I think we're ready to ship this!

Commit:
Making sure labels and titles are consistent

Should be external dataset webpage, not dataset page or landing page.

zednis commented:
:shipit:
